### ISSUE TITLE
camel_case_killer conflicts with conversion_box and default_box

### ISSUE THREAD

**Issue by matan129** [cdgriffith/Box#46]
Hey!
I've recently started using your awesome package and it has been a delight. Unfortunately, I encountered couple of issues with the `camel_case_killer` option:
![image](https://user-images.githubusercontent.com/1483099/35511821-1a6a47d0-0506-11e8-932a-af350abd1de4.png)

I've tested it on Python 2 and 3. 
Copy-pasteable reproduction code:
```python
from box import Box

b = Box({
    'SomeKey': 999
}, camel_killer_box=True, conversion_box=True)

assert b.some_key == 999 # Cool, works as expected
assert b.SomeKey == 999  # Should not work because of conversion_box=True!

# When using default_box=True, things get even worse:
b = Box({
    'SomeKey': 999
}, camel_killer_box=True, conversion_box=True, default_box=True)

assert b.SomeKey == 999   # Same issue
assert b.some_key == Box({})  # Oh man, the default_box is prioritized over camel_killer_box...
```

**Comment by cdgriffith** [comment]
Hey Matan, just wanted to say thanks for laying out the issues you found so clearly and for the great PR! (You're also the first person to fully follow the contributor guidelines, much appreciated.) 

When I have a moment today I will give it a few pokes myself and merge your PR 😄

**Comment by cdgriffith** [comment]
Just wanted to add that even with both `conversion_box` and `camel_killer_box` an attribute added like `SomeAttr` would become accessible `some_attr`, but still hold the key in the original space of `SomeAttr`, it is simply a handy way to access it, it does not change the underlying dict object.

**Comment by matan129** [comment]
Okay, if it's really the intended behavior, the documentation is misleading:
![image](https://user-images.githubusercontent.com/1483099/35589164-931cc8c8-060b-11e8-9c36-2e0d0f4228b5.png)

**Comment by cdgriffith** [comment]
You are correct, either doc or my current thinking is definitely wrong. Thanks for pointing that out. I will have to dig into when I added that and original thinking and make sure I'm not forgetting something. (Because at least right now my thinking is that the underlying dict should never be modified beyond transformation into a Box object. )
